src/bootstrap.php

```diff
@@ -202,7 +202,9 @@
 
     $recommendations = $relations;
 
-    $appendFirstThatDoesNotAlreadyExist = function (Sequence $recommendations, Sequence $toInsert) : Sequence {
+    $appendFirstThatDoesNotAlreadyExist = function (Sequence $recommendations, Sequence $toInsert, int $max = 1) : Sequence {
+        $found = 0;
+
         foreach ($toInsert as $item) {
             foreach ($recommendations as $recommendation) {
                 if (
```

Comment:
perhaps it's time to extract this into a `equals()` private method since its acrobatic implementation can be distracting in the context of merging part of a `Sequence` into another

Reply:
or `containsAlready($recommendations, $item)`

Reply:
or `alreadyExists($item, $recommendations)` which is consistent with the rest